actor-server/actor-http-api/src/main/scala/im/actor/server/api/http/HttpApiFrontend.scala

```diff
@@ -26,7 +23,16 @@ final class HttpApi(_system: ActorSystem) extends Extension {
   implicit val ec = system.dispatcher
   implicit val mat = ActorMaterializer()
 
-  val hooks = new HttpApiHookControl
+  private val hooks = new HttpApiHookControl
+
+  def runHooks(): Future[Seq[Route]] = hooks.routesHook.runAll()
+
+  // should we pass implicit actor system here, or the one we have in implicit scope here will work???
+  def registerHook(name: String)(f: ActorSystem ⇒ Future[Route]): Unit = {
```

Comment:
We always use it in following way: `HttpApi(system).registerHook()`, and extension is always initialized with system, so we can use same system in registerHook, right?